vms/platformvm/warp/message/validator_uptime.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package message
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+// ValidatorUptime is signed when the ValidationID is known and the validator
+// has been up for TotalUptime seconds.
+type ValidatorUptime struct {
+	payload
```

Comment:
What is this embed for + why do we need this interface?

Reply:
As for interface, what interface are you referring to? `ValidatorUptime` is a struct.

Reply:
> As for interface, what interface are you referring to? ValidatorUptime is a struct.

We're embedding `payload` - it seems that the reason we're doing this is to implement the `Payload` interface. I'm trying to figure out exactly why we need to use this type.

> This is cleaner, and clearly the AvalancheGo approach

We should try to understand what the `Payload` interface is actually used for before copying it into our usage here - our rationale of being clean/previously existing isn't a technical reason. Why does the `Payload` interface need to exist in the P-Chain, and do we need it here? I suspect that this is closely related to our discussion over us sharing codecs.